### PR TITLE
Add `pages` proc option

### PR DESCRIPTION
**Pull request by hiro-koba**
Added a feature to process per page.
I sometimes face a situation that I need to process records in batch.

### CODE REVIEW

File: lib/embulk/filter/ruby_proc.rb [hunk 1]
```diff
@@ -207,10 +235,22 @@ def add(page)
...
             end
           end
         end
+
+        unless page_procs.empty?
+          tmp_records = page_procs.each_with_object([]) do |pr, arr|
+            result = pr.call(proc_records)
+            result.each { |r| arr << r }
+          end
+          tmp_records.each { |record| page_builder.add(record.values) }
```
Comment (joker1007) [inline lib/embulk/filter/ruby_proc.rb]: Do you have any reason why you wrote `each` loop twice?

File: lib/embulk/filter/ruby_proc.rb [hunk 2]
```diff
@@ -85,6 +87,13 @@ def self.transaction(config, in_schema, &control)
             eval(File.read(rowdef["proc_file"]), evaluator_binding, File.expand_path(rowdef["proc_file"]))
           end
         }.compact
+        @page_proc_store[transaction_id] = page_procs = task["pages"].map {|rowdef|
```
Comment (joker1007) [inline lib/embulk/filter/ruby_proc.rb]: I think that `rowdef` is not suiltable name for `pages` definition.

File: lib/embulk/filter/ruby_proc.rb [hunk 3]
```diff
@@ -146,19 +161,31 @@ def self.parse_row_procs(rows, evaluator_binding)
         }.compact
       end
 
+      def self.parse_page_procs(rows, evaluator_binding)
+        rows.map {|rowdef|
```
Comment (joker1007) [inline lib/embulk/filter/ruby_proc.rb]: I think that `rows` and `rowdef` are not suiltable name for `parse_page_procs` method.

**Review by joker1007 — COMMENTED** [review]
Thanks you so much :+1: 
I commented about some points.

File: lib/embulk/filter/ruby_proc.rb (hunk 2 above)
Comment (hiro-koba) [reply lib/embulk/filter/ruby_proc.rb]: Oops, sorry for the copy and paste code... Fixed it.

File: lib/embulk/filter/ruby_proc.rb (hunk 1 above)
Comment (hiro-koba) [reply lib/embulk/filter/ruby_proc.rb]: I removed an unnecessary loop. Thank you.

**Comment by hiro-koba** [comment]
@joker1007 Thanks for the quick response! I fixed all of your reviews.

**Review by joker1007 — APPROVED** [review]
LGTM

**Comment by joker1007** [comment]
Thank you for this patch!!

**Pull request merged**